Framework/API/src/AlgorithmFactory.cpp

```diff
@@ -382,8 +410,17 @@ AlgorithmFactoryImpl::getDescriptors(bool includeHidden) const {
         currentLayer.append("\\");
       }
 
-      if (!categoryIsHidden)
+      if (!categoryIsHidden) {
         res.emplace_back(desc);
+        // Add alias to results if included
+        if (!desc.alias.empty() && includeAliases) {
+          AlgorithmDescriptor aliasDesc;
```

Comment:
C++11 allows [aggregate initialization](https://gcc.godbolt.org/z/zaj7Ev) so I think we can reduce this to
```c++
res.emplace_back({desc.alias, desc.version, desc.category, ""});
```

and this will avoid having to create a temporary object and pushing it in.

Reply:
The change as you have requested produces an error, I am not familiar with this style. I tried `res.emplace_back(AlgorithmDescriptor{desc.alias, desc.version, desc.category, ""});` instead and it builds fine. Am I right in assuming this is the correct syntax?

Reply:
@Matt-Cumber I thought `emplace_back` could cope with that but apparently not. Your version is fine as it still avoids the temp and moves into place.